source/microbit/microbitaccelerometer.cpp

```diff
@@ -36,28 +36,57 @@ typedef struct _microbit_accelerometer_obj_t {
     MicroBitAccelerometer *accelerometer;
 } microbit_accelerometer_obj_t;
 
+bool accelerometer_up_to_date = false;
+
 mp_obj_t microbit_accelerometer_get_x(mp_obj_t self_in) {
     microbit_accelerometer_obj_t *self = (microbit_accelerometer_obj_t*)self_in;
+    if (!accelerometer_up_to_date) {
+        self->accelerometer->idleTick();
```

Comment:
Good point, I've made the accelerometer and compass update functions safe against interrupt.